### ISSUE TITLE
Есть поля которые нельзя сопоставить с Logs API Метрики

### ISSUE THREAD

**Issue by vodmanGithub** [zhdanchik/yandex_metrika_connector_cases#2]
Необходимые поля из вашего файла:

> CounterID, UserIDHash, VisitID, VisitVersion, Sign, 
> StartDate, UTCStartTime, 
> `TrafficSource.Model`, `TrafficSource.StartTime`,  
> `TrafficSource.ID`, `TrafficSource.SearchEngineID`, 
> `TrafficSource.AdvEngineID`, `TrafficSource.SocialSourceNetworkID`, 
> `TrafficSource.ClickBannerID`, `TrafficSource.ClickTargetType`, 
> `TrafficSource.RecommendationSystemID`, `TrafficSource.MessengerID`, 
> `EPurchase.ID`, `EPurchase.Revenue`,
> `Goals.ID`, `Goals.Price`

Как их сопоставить с [названиями полей](https://yandex.ru/dev/metrika/ru/logs/fields/visits) из документации LogsAPI?

**Comment by zhdanchik** [comment]
Давай попробуем!

**Поля as is**
- `CounterID` == `ym:s:counterID`
- `UserIDHash` == `ym:s:counterUserIDHash`
- `VisitID` == `ym:s:visitID`
- `StartDate` == `ym:s:date`
- `UTCStartTime` == `ym:s:dateTimeUTC`
- `EPurchase.ID` == `ym:s:purchaseID`
- `EPurchase.Revenue` == `ym:s:purchaseRevenue`
- `Goals.ID` == `ym:s:goalsID`
- `Goals.Price` == `ym:s:goalsPrice`

**Поля, которых нет в LogsAPI по причине отсуствия там версионирования** (и это ок, можно взять просто что-то, что не будет никак влиять на конечный результат)
- `VisitVersion` == `1`
- `Sign` == `1`

**Поля атрибуций, в logsapi возвращаются отдельными полями, а в коннекторе массивами** (для работы с цепочками нам нудны только last-click атрибуции можно из них создать массивы длинной 1, и в Model положить единичку
- `TrafficSource.Model` == `[1]`
- `TrafficSource.StartTime` == `[ym:s:dateTime]`

**Тоже поля атрибуций с названиями источников, всместо ID** (я использую ID и их преобразовывание в названия по коду, тут же уже есть названия, которые можно использовать по коду, но придется немного видоизменять код тоже)

- `TrafficSource.ID` ~= `[ym:s:lastTrafficSource]`
- `TrafficSource.SearchEngineID` ~= `[ym:s:lastSearchEngine]`
- `TrafficSource.AdvEngineID` ~= `[ym:s:lastAdvEngine]`
- `TrafficSource.SocialSourceNetworkID` ~= `[ym:s:lastSocialNetwork]`
- `TrafficSource.RecommendationSystemID` ~= `[ym:s:lastRecommendationSystem]`
- `TrafficSource.MessengerID` ~= `[ym:s:lastMessenger]`

**Поле кликов, которое нужно для того, чтобы понять, что клик приклеен** (подойдет и стринговая версия, но в коде нужно буджет изменить проверку с != 0 на != '')
- `TrafficSource.ClickBannerID` ~= `[ym:s:lastDirectClickBanner]`

**Поле кликов, которое нужно для того, чтобы разделить поисковую рекламу яндекса от РСЯ** (тут тоже в logsapi нет числового поля, нужно будет брать текстовое и жить с ним)
- `TrafficSource.ClickTargetType` ~= `[ym:s:lastDirectPlatformType]`

т.е. резюмируя, сопоставить можно, но код немного нужно будет подхачить

**Comment by vodmanGithub** [comment]
Можно сделать просто отдельный файл для стандартного набора полей LogsAPI, если решитесь порефакторить

**Comment by zhdanchik** [comment]
Да, пожалуй можно написать какой-то "адаптер" данных LogsAPI для этой техники. Времени особо не было это сделать перед выкладкой, запланирую это сделать в этом году.

**Comment by artbog** [comment]
@zhdanchik сделать в этом году еще планируете или уже на следующий перенесете?
